packages/backend-test-utils/src/services/mockServices.ts

```diff
@@ -538,4 +539,20 @@ export namespace mockServices {
       subscribe: jest.fn(),
     }));
   }
+
+  export namespace systemMetadata {
+    /**
+     * Creates a functional mock factory for the
+     * {@link @backstage/backend-plugin-api#coreServices.systemMetadata}.
+     */
+    export const factory = () => systemMetadataServiceFactory;
```

Comment:
Thinking it might be best to have a proper mock implementation here that doesn't attempt network calls? Ideally it's even configurable to be able to try different discovered features, but can leave that as a future addition too.